### DISCUSSION TITLE
How to programmatically get the "launched by" information for a job

### DISCUSSION THREAD

**Discussion by dhirschfeld** [dagster-io/dagster#17801] [General]
If I have the `context`/`run_id` of a job, how can I get the "Launched by" information for taht run, as shown in the UI:

![image](https://github.com/dagster-io/dagster/assets/881019/54e1d2b3-10bb-4c30-ba1e-7fe166a22c18)

i.e. I'm after the user (or schedule) that triggered the run

**Comment by dhirschfeld** [comment]
I tried `get_run_stats` to no avail :(
```
 In[39]: context.instance.get_run_stats(context.dagster_run.run_id)._asdict()
Out[39]: 
{'run_id': '44c0949d-b29f-43dc-839e-9f7f35617614',
 'steps_succeeded': 1,
 'steps_failed': 0,
 'materializations': 1,
 'expectations': 0,
 'enqueued_time': None,
 'launch_time': 1699407579.09803,
 'start_time': 1699407587.283498,
 'end_time': None}

```

**Reply by logantran26** [reply]
I believe "Launched by" from the UI is from the tag `dagster/schedule_name` of the job. When it is "Manually launched" it means that this tag is not found. I may be wrong though. But I tried that in my sensor and it worked. My sensor only tracks a particular job and it will only trigger when this job is not manually launched. The context.log.info is there for testing purpose

```
daily_dbt_models_job = JobSelector(location_name='orchestration-dbt', job_name='daily_dbt_models')

@run_status_sensor(
    run_status=DagsterRunStatus.SUCCESS
    ,minimum_interval_seconds=600
    ,monitored_jobs=[daily_dbt_models_job]
    ,request_job=dagster_graphql_daily_job
)
def dagster_graphql_sensor(context):
    partition_date = s3_utils.convert_date(datetime.now(timezone.utc) - timedelta(days=1), '%Y-%m-%d')
    run_tags = context.dagster_run.tags
    schedule_name = run_tags.get('dagster/schedule_name', '') #Only scheduled runs have this tag, ensure the sensor is not triggered if this job is run manually in the launchpad
    if schedule_name == 'daily_dbt_models_schedule':
        context.log.info('Executing for scheduled run.')
        return RunRequest(
            run_key=None
            ,partition_key=partition_date
        )
    else:
        context.log.info('Skipping for manual run.')
    return None
```

**Comment by jamiedemaria** [comment]
The launcher info is stored in tags. You can get the tags for a run by calling `instance.get_run_tags()`. If it's a schedule it'll have `dagster/schedule_name` tag

**Reply by C0DK** [reply]
Is this still the case? in the OSS version no specific user is defined, and by looking at the tags I get in my current manually triggered run on my local machine, doesn't have any such key. I get `.dagster/grpc_info`, `dagster/parent_run_id`, `dagster/partition`, `dagster/partition_set`, `dagster/root_run_id`.

**Reply by jamiedemaria** [reply]
sorry about that, i misread some code. for OSS you won't be able to get a user tag since we don't have a concept of multiple users for a single deployment in OSS. I updated the original answer to reduce confusion in the future

**Reply by C0DK** [reply]
I'd hope there was still some sort of source, but my assumption is that if neither auto materialization nor schedule is defined, then it's manually launched (in the OSS version)

**Reply by ion-elgreco** [reply]
We are currently exploring if we can fetch the username from the oauthproxy session and then pass this to the custom run executor.

To mark it as run by you need to add a tag: {'user':'john doe'}

**Reply by C0DK** [reply]
i didn't really care for who did it, merely the type of run. I wanted to use it for our alerting logic, so (for instance) alerts were raised with a lower criticality  on manually started jobs. :)

**Reply by temminks** [reply]
@ion-elgreco did you get this to work or found another way?

**Reply by ion-elgreco** [reply]
@temminks yes this works, you can fetch the headers with a customRunExecutor

**Comment by emanueledomingo** [comment]
Hi guys.

I think the issue i'm encountering it's related to this: in the OSS version, can i set this "Launched By" metadata programmatically? (for example by setting a tag, as i already do for partitions in some use cases)
